examples/advanced/federated-statistics/hierarchical_stats/demo/visualization.ipynb

```diff
@@ -0,0 +1,222 @@
+    "    data = json.load(f)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1b0f21fd",
+   "metadata": {},
+   "source": [
+    "### Overall Statistics\n",
+    "\n",
+    "In this section, we demonstrate how to visualize hierarchical stats from the tabular data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ab771712",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "def create_widgets_hierarchy(data, parent_key='', sep='_'):\n",
+    "    \"\"\"Convert hierarchical JSON into nested widgets for display.\"\"\"\n",
+    "    if isinstance(data, dict):\n",
+    "        items = []\n",
+    "        for k, v in data.items():\n",
+    "            if k == \"Global\" or k == \"Local\":\n",
+    "                flattened_data = []\n",
+    "                for metric, values in v.items():\n",
+    "                    for key, value in values[\"default_set\"].items():\n",
+    "                        flattened_data.append({\n",
+    "                            \"Metric\": metric,\n",
+    "                            \"Type\": key,\n",
+    "                            \"Value\": value\n",
+    "                        })\n",
+    "                df = pd.DataFrame(flattened_data)\n",
+    "                df_pivot = df.pivot(index=\"Metric\", columns=\"Type\", values=\"Value\").T\n",
+    "                output = widgets.Output()\n",
+    "                with output:\n",
+    "                    display(df_pivot)\n",
+    "                items.append((k, output))\n",
+    "            elif k == \"Name\":\n",
+    "                continue\n",
+    "            else:\n",
+    "                new_key = f\"{parent_key}{sep}{k}\" if parent_key else k\n",
+    "                items.append((k, create_widgets_hierarchy(v, new_key, sep=sep)))\n",
+    "        return widgets.Accordion(children=[v for k, v in items], titles=[k for k, v in items])\n",
+    "    elif isinstance(data, list):\n",
+    "        items = []\n",
+    "        for i, item in enumerate(data):\n",
+    "            new_key = f\"{parent_key}{sep}{i}\" if parent_key else str(i)\n",
+    "            items.append((item.get(\"Name\"), create_widgets_hierarchy(item, new_key, sep=sep)))\n",
+    "        return widgets.Accordion(children=[v for k, v in items], titles=[k for k, v in items])\n",
+    "    else:\n",
+    "        return widgets.Text(value=str(data), disabled=True)\n",
+    "\n",
+    "\n",
+    "widget_hierarchy = create_widgets_hierarchy(data)\n",
+    "display(widget_hierarchy)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "95e42829",
+   "metadata": {},
+   "source": [
+    "### Histogram Visualization\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3e86860e",
+   "metadata": {},
+   "source": [
+    "The following code demonstrates how the hierarchical histograms can be displayed display for for features like `percentage`."
```

Comment:
suggest "The following code demonstrates how the hierarchical histograms can be displayed using Jupyter Widgets for features like `percentage`."

Reply:
Done, thanks.